src/zh/roumanwu/src/eu/kanade/tachiyomi/extension/zh/roumanwu/Roumanwu.kt

```diff
@@ -36,41 +42,100 @@ class Roumanwu : HttpSource(), ConfigurableSource {
+        title = element.select("div.hidden > div.px-2 > div.truncate").text()
+        url = element.attr("href")
+        thumbnail_url = element.select("div.bg-cover").attr("style").substringAfter("background-image:url(\"").substringBefore("\")")
+    }
 
-    override fun mangaDetailsParse(response: Response) = response.nextjsData<BookDetails>().book.toSManga()
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        title = document.select("main > div:first-child > div > div.grid > div:first-child div.text-xl").text()
+        thumbnail_url = baseUrl + document.select("main > div:first-child img").attr("src")
+        author = document.select("main > div:first-child > div > div.grid > div:first-child > div:first-child > div:nth-child(2) > div:nth-child(3) span").text()
+        artist = author
+        genre = document.select("main > div:first-child > div > div.grid > div:first-child > div:first-child > div:nth-child(2) > div:nth-child(6) span").text().replace(",", ", ")
+        description = document.select("main > div:first-child > div > div.grid > div:first-child > div:nth-child(2)").text().substring(3)
+    }
+
+    override fun chapterListSelector(): String = "main > div:first-child > div > div.grid > div:first-child > div:nth-child(5) > a"
+    override fun chapterFromElement(element: Element): SChapter = SChapter.create().apply {
+        url = element.attr("href")
+        name = element.text()
+    }
+    override fun chapterListParse(response: Response): List<SChapter> {
+        return super.chapterListParse(response).reversed()
+    }
 
-    override fun chapterListParse(response: Response) = response.nextjsData<BookDetails>().book.getChapterList().reversed()
+    override fun pageListParse(document: Document): List<Page> {
+        val jsonString = document.select("script")
+            .find { it.html().contains("imageUrl") }
+            ?.html()
+            ?.let { content ->
+                """(?<=\[1,").*(?="\])""".toRegex()
+                    .find(content)
+                    ?.value
+                    ?.substring(2)
+                    ?.dropLast(2)
+                    ?.replace("\\\"", "\"")
+            }
 
-    override fun pageListParse(response: Response): List<Page> {
-        val chapter = response.nextjsData<Chapter>()
-        if (chapter.statusCode != null) throw Exception("服务器错误: ${chapter.statusCode}")
-        return if (chapter.images != null) {
-            chapter.getPageList()
-        } else {
-            @Suppress("NAME_SHADOWING")
-            val response = client.newCall(GET(baseUrl + chapter.chapterAPIPath!!, headers)).execute()
-            if (!response.isSuccessful) throw Exception("服务器错误: ${response.code}")
-            response.parseAs<ChapterWrapper>().chapter.getPageList()
-        }
+        return jsonString?.let { str ->
+            val json = Json.parseToJsonElement(str)
```

Comment:
```suggestion
            val jo = json.parseToJsonElement(str)
```
use app provided `json` instance